packages/phisherman/src/lib/Phisherman.ts

```diff
@@ -0,0 +1,90 @@
+import fetch, { Headers } from 'node-fetch';
+import { fromAsync, isErr } from '@sapphire/framework';
+
+export class Phisherman {
+	/**
+	 * The options for phisherman
+	 */
+	public options;
+	public constructor(options: PhishermanOptions | undefined) {
+		this.options = options;
+		void this.checkApiKey(this.options!.apiKey).then((res) => {
+			if (res) throw new Error('Invalid API key');
+		});
+	}
+
+	/**
+	 * Check if a link is scam
+	 * @param domain The link to check
+	 * @since 1.0.0
+	 */
+	public async check(domain: string) {
+		const result = await fromAsync(async () => {
+			const check = fetch(`https://api.phisherman.gg/v2/domains/check/${domain}`, {
+				headers: new Headers({
+					'Content-Type': 'application/json',
+					Authorization: `Bearer ${this.options?.apiKey}`
+				})
+			}).then((body) => body.json());
+			return check as Promise<PhishermanReturnType>;
```

Comment:
```suggestion
			return await fetch<PhishermanReturnType>(`https://api.phisherman.gg/v2/domains/check/${domain}`, {
				headers: {
					'Content-Type': 'application/json',
					Authorization: `Bearer ${this.options?.apiKey}`
				}
			}, FetchResultTypes.JSON);
```

After changing fetch to @sapphire/fetch